### PR TITLE
File request API

### PR DESCRIPTION
**Pull request by cbetta**
# Description

File request API. Cleaned up copy of #114 

RE: APIWG-210, DDOC-372

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own changes
- [X] I have run `yarn lint` to make sure my changes pass all linters
- [X] I have pulled the latest changes from the upstream developer branch

Schedule: 9/28

### CODE REVIEW

**Comment by cbetta** [comment]
@virenderg made some big changes. Would love your eyes on it. Should be good to merge as is.

File: content/requests/file_request_update_request.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+      This will default to the value on the existing file request.
+    example: "Please upload required documents"
+
+  status:
+    type: string
+    description: |-
+      An optional new status of the file request.
+
+      When the status is set to `INACTIVE`, the file request
```
Comment (cbetta) [inline content/requests/file_request_update_request.yml:32]: @virenderbox did we not agree on using lower case values for this in APIWG?

**Comment by cbetta** [comment]
Preview https://staging.developer.box.com/reference/resources/file-request/

File: content/requests/file_request_update_request.yml (hunk 1 above)
Comment (virenderg) [reply content/requests/file_request_update_request.yml:33]: We discussed in APIWG. However, decided to keep them this way as our V3 API uses upper case and is currently live. If this is a requirement, we can change it.

Comment (cbetta) [reply content/requests/file_request_update_request.yml:33]: The V2 styleguide uses lowercase for these kind of status strings and we should not deviate if we can fix this at the V2 level.

Comment (virenderg) [reply content/requests/file_request_update_request.yml]: Updated.

File: content/responses/file_request.yml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    readOnly: true
+
+  etag:
+    type: string
+    example: "1"
+    nullable: true
+    description: |-
+      The HTTP `etag` of this file. This can be used within some API
+      endpoints in the `If-Match` and `If-None-Match` headers to only
+      perform changes on the file request if (no) changes have happened.
```
Comment (cbetta) [inline content/responses/file_request.yml:129]: @virenderg do any of the new APIs support the `If-Match`/`If-None-Match` headers? If so, I can add them to those endpoints.

Comment (virenderg) [reply content/responses/file_request.yml:129]: @cbetta , Yes the PUT endpoint. I added If-Match and 412 error codes in previous commit (https://github.com/box/box-openapi/blob/1040e562a665a94954b942c219836bb491604d28/content/paths/file_requests__put_file_requests_id.yml#L19). Let me know if other changes are needed.

Comment (cbetta) [reply content/responses/file_request.yml:129]: Ah you already did. I simplified the language here a bit so it is a bit more specific to this usecase, and only mentions the `If-Match` header (cause I doubt the `If-None-Match` header makes sense here).

**Pull request merged**